### ISSUE TITLE
Expose numba -s via python -m and in Jupyter

### ISSUE THREAD

**Issue by cdeil** [numba/numba#4038]
Suggest to expose the `numba` cli also via `python -m numba` using https://docs.python.org/3/library/__main__.html 
Then it's easier to access in cases where PATH isn't set correctly, like what seems to be the case  in #3987 .

In addition (or maybe it's the same thing) I think it would be useful to be able to print out the `numba -s` info from inside Python/ipython/Jupyter. Personally I often use Jupyter and have many conda envs, and get confused when checking performance / timings from Jupyter notebooks - being able to print the `numba -s` info into the notebook would help.
(For me `!numba -s` doesn't work, numba isn't on the path, not sure why yet)

**Comment by esc** [comment]
I'd like to add, that exporting it as JSON might be advantageous in case you wish to feed the output to another tool to consume.

**Comment by stuartarchibald** [comment]
Thanks for the request. This ticket is related: https://github.com/numba/numba/issues/2459, if:
https://github.com/numba/numba/blob/e8ac4951affacd25c63ba2c18d62a3f12ed7e0ba/numba/numba_entry.py#L9
returned a dictionary then it's more use for programmatic consumption and trivial to convert to JSON or YAML etc.

**Comment by cdeil** [comment]
What about the suggestion to add a `numba/__main__.py` to make `python -m numba` work (in addition to the current `numba` cli that's created on install?

I think it would be useful for users that have a broken install, or devs that work with an in-place build as suggested (http://numba.pydata.org/numba-doc/latest/developer/contributing.html#building-numba) - there `numba` cli wont' be available but one could still execute `python -m numba`.

If you agree `numba/__main__.py` is a useful addition, I could send a PR for that.

Turns out that `!numba -s` usually works in Jupyter notebooks, I just got unlucky and tried it in a broken installation (reported in #4040). Still probably worth exposing and documenting how to get the info via a function call.

I could also take a stab at #2459 - but that's a larger edit / should be a separate PR.
